compiler/rustc_passes/src/dead.rs

```diff
@@ -647,6 +650,29 @@ impl<'tcx> Visitor<'tcx> for MarkSymbolVisitor<'tcx> {
 
         self.in_pat = in_pat;
     }
+
+    fn visit_poly_trait_ref(&mut self, t: &'tcx hir::PolyTraitRef<'tcx>) {
+        // Mark the assoc const appears in poly-trait-ref live
+        if let Some(pathsegment) = t.trait_ref.path.segments.last()
+            && let Some(args) = pathsegment.args
+        {
+            for constraint in args.constraints {
+                if let Some(item) = self
+                    .tcx
+                    .associated_items(pathsegment.res.def_id())
+                    .filter_by_name_unhygienic(constraint.ident.name)
```

Comment:
I'm sufficiently sure that this name resolution is already performed somewhere else, and the result is stored in some table.
So here we need to obtain it from that table by id, instead of doing resolution by name again.

Reply:
Using `find_by_ident_and_kind` now and I think it's better than the old version.

Although `find_by_ident_and_kind` will also call `filter_by_name_unhygienic` and then does the similar thing, but I find the impl of `filter_by_name_unhygienic` is doing the filtering by obtaining assoc items from `AssocItems.items: SortedIndexMultiMap` by using the given ident as the key in fact. Maybe this is the table you mentioned.

Reply:
Ok, I'm still sure there's a better way, but this is good enough.